src/messages.rs

```diff
@@ -363,3 +364,169 @@ pub enum ResponseContent {
+                write!(formatter, "ResponseContent {{ GetNetworkName {{ {:?} }} }}", relocated_id),
+            ResponseContent::GetPublicId { ref public_id } =>
+                write!(formatter, "ResponseContent {{ GetPublicId {{ {:?} }} }}", public_id),
+            ResponseContent::GetPublicIdWithEndpoints { ref public_id, .. } =>
+                write!(formatter, "ResponseContent {{ GetPublicIdWithEndpoints {{ {:?}, .. }} }}", public_id),
+            ResponseContent::GetCloseGroup { ref close_group_ids } =>
+                write!(formatter, "ResponseContent {{ GetCloseGroup {{ {:?} }} }}", close_group_ids),
+            ResponseContent::GetSuccess(ref data, ref message_id) =>
+                write!(formatter, "ResponseContent {{ GetSuccess {{ {:?}, {:?} }} }}", data, message_id),
+            ResponseContent::PutSuccess(ref digest, ref message_id) =>
+                write!(formatter, "ResponseContent {{ PutSuccess {{ {:?}, {:?} }} }}", digest, message_id),
+            ResponseContent::PostSuccess(ref digest, ref message_id) =>
+                write!(formatter, "ResponseContent {{ PostSuccess {{ {:?}, {:?} }} }}", digest, message_id),
+            ResponseContent::DeleteSuccess(ref digest, ref message_id) =>
+                write!(formatter, "ResponseContent {{ DeleteSuccess {{ {:?}, {:?} }} }}", digest, message_id),
+            ResponseContent::GetFailure { ref id, ref request, .. } =>
+                write!(formatter, "ResponseContent {{ GetFailure {{ {:?}, {:?}, .. }} }}", id, request),
+            ResponseContent::PutFailure { ref id, ref request, .. } =>
+                write!(formatter, "ResponseContent {{ PutFailure {{ {:?}, {:?}, .. }} }}", id, request),
+            ResponseContent::PostFailure { ref id, ref request, .. } =>
+                write!(formatter, "ResponseContent {{ PostFailure {{ {:?}, {:?}, .. }} }}", id, request),
+            ResponseContent::DeleteFailure { ref id, ref request, .. } =>
+                write!(formatter, "ResponseContent {{ DeleteFailure {{ {:?}, {:?}, .. }} }}", id, request),
+        }
+    }
+}
+
+
+#[cfg(test)]
+mod test {
+    extern crate rand;
+
+    use super::{HopMessage, SignedMessage, RoutingMessage, RequestMessage, RequestContent};
+    use id::FullId;
+    use authority::Authority;
+    use xor_name::XorName;
+    use sodiumoxide::crypto::sign;
+
+    #[test]
+    fn signed_message_check_integrity() {
+        let name: XorName = rand::random();
+        let routing_message = RoutingMessage::Request(
+            RequestMessage {
+                src: Authority::ClientManager(name),
+                dst: Authority::ClientManager(name),
+                content: RequestContent::Connect,
+            }
+        );
+        let full_id = FullId::new();
+        let signed_message_result = SignedMessage::new(routing_message.clone(), &full_id);
+
+        assert!(signed_message_result.is_ok());
+
+        let signed_message = signed_message_result.unwrap();
```

Comment:
Can you replace the bare `unwrap` with our `unwrap_result` macro please?  There are a few instances in these tests.